roadmap/state-sync-protocol.md

```diff
@@ -0,0 +1,198 @@
+# Execution State Synchronization Protocol
+
+## Context
+
+Today, Access nodes act as the gateway to the Flow network, providing a query endpoint for users (REST, gRPC) and performing important admission control duties such as:
+* Rate limiting of requests (script executions, Collection requests, account queries, etc.)
+* Basic validation, preventing bad transactions from entering the network
+
+Execution nodes and Collection nodes currently still expose public APIs, vestiges of an earlier version of the protocol in which Access nodes had not yet been introduced. However, allowing users to bypass the Access layer and make direct requests to these nodes not only dilutes the valuable resources which should be reserved for performing tasks crucial to the protocol (especially in the case of Execution nodes), but it also leaves them at risk of denial-of-service attacks. Going forward, in order to protect the other staked nodes of the network, the Access layer will become the *only* way for users to interact with the network.
```

Comment:
I feel like theres some inaccuracies here.

> vestiges of an earlier version of the protocol in which Access nodes had not yet been introduced

I'm not sure if this is the right way to think about this. Even after adding the access layer, we were still adamant about being able to interact with the chain through Collection and Execution nodes, as a feature. This section in the doc seems implies that it was an unwanted side effect.

> the Access layer will become the *only* way for users to interact with the network.

Also, this is quite contentious at the moment, so I'm not so sure it makes sense to have it here before a decision is made.

Reply:
> Execution nodes and Collection nodes currently still expose public APIs, vestiges of an earlier version of the protocol in which Access nodes had not yet been introduced.

Access nodes were already part of the architecture _before_ we started with the Go implementation. The reasons why we started with APIs on the ENs and Collectors was simply a short-cut. 

```suggestion
Execution nodes and Collection nodes currently still expose public APIs, vestiges of earlier short cuts. However, allowing users to bypass the Access layer and make direct requests to these nodes not only dilutes the valuable resources which should be reserved for performing tasks crucial to the protocol (especially in the case of Execution nodes), but it also leaves them at risk of denial-of-service attacks. Going forward, in order to protect the other staked nodes of the network, the Access layer will become the *only* way for users to interact with the network.
```

Reply:
> the Access layer will become the only way for users to interact with the network.

agree with Kan. I think the sync engine will be needed either way, no matter whether the EN or the AN exposes the API. I would argue as follows:

> It is important for network-external entities as well as network-internal nodes to be able to acquire a full copy of the execution state. This is important to remove load from the Execution node. Furthermore, due to their (intentionally) low number, entities other than the Execution Nodes should be able to hold (and potentially archive) the full state to further decentralization.

Reply:
> Also, this is quite contentious at the moment, so I'm not so sure it makes sense to have it here before a decision is made.

yeah, this was written before we had our previous discussion on Access nodes, and I haven't had time to update it since then